### PR TITLE
Forman xxx xcube intergr 2

### PR DESCRIPTION
**Pull request by forman**
### Changes 2.3.0 (in development)

* Cate App 3.0 now requires the Web API service of the `cate 3.0+` 
  Python package.

* In order to keep alive the connection to the Web API service,
  Cate App now sends a keepalive signal every 2.5 seconds. (#150)

* Optimisations in the DATA SOURCES panel (that have been enabled by 
  using [xcube](https://xcube.readthedocs.io/) in the backend):
  - Initalising the "ESA CCI Open Data Portal" data store
    is now accellerated by a magnitude.
  - Local caching of remote data sources when opening datasets 
    is now much faster and more reliable.
  - Ability to add more data stores has been greatly improved.

* We now obtain Cate Hub's status information from a dedicated GitHub 
  repository [cate-status](https://github.com/CCI-Tools/cate-status).

### CODE REVIEW

File: src/renderer/actions.ts [hunk 1]
```diff
@@ -41,10 +41,10 @@ import * as selectors from './selectors';
 import {
     BackendConfigState,
     ColorMapCategoryState,
-    ControlState,
+    ControlState, DatasetDescriptor,
```
Comment (TonioF) [inline src/renderer/actions.ts:44]:
```suggestion
    ControlState, 
    DatasetDescriptor,
```

File: src/renderer/actions.ts [hunk 2]
```diff
@@ -41,10 +41,10 @@ import * as selectors from './selectors';
...
     DataSourceState,
     DataStoreState,
-    GeographicPosition,
+    GeographicPosition, HubStatus,
```
Comment (TonioF) [inline src/renderer/actions.ts:47]:
```suggestion
    GeographicPosition, 
    HubStatus,
```

File: src/renderer/components/DataSourceItem.tsx [hunk 3]
```diff
@@ -81,4 +68,35 @@ const DataSourceItem: React.FC<DataSourceItemProps> = ({dataSource, showDataSour
...
+
+function dataSourceToTextIconProps(dataSource: DataSourceState) {
+    let ecvId;
+    let label;
+    if (dataSource.title) {
+        ecvId = dataSource.title.split(' ', 1)[0].toLowerCase();
+        label = dataSource.title.substr(0, 3).toUpperCase();
+    }
+    if (!ecvId || !ECV_META.ecvs[ecvId]) {
+        // This is a CCI-store specific hack
```
Comment (TonioF) [inline src/renderer/components/DataSourceItem.tsx:82]: It there is a hack for the CCI-Store, we should consider using one for the pilot store as well.

**Review by TonioF — APPROVED** [review]
Only minor comments.

File: src/renderer/components/DataSourceItem.tsx (hunk 3 above)
Comment (forman) [reply src/renderer/components/DataSourceItem.tsx:82]: > It there is a hack for the CCI-Store [sic]

If we preserve the `ecvId` in the Zarrs then this should work too!

**Pull request merged**